src/MissionManager/VisualMissionItem.cc

```diff
@@ -162,11 +162,11 @@ void VisualMissionItem::setMissionVehicleYaw(double vehicleYaw)
 
 void VisualMissionItem::_updateTerrainAltitude(void)
 {
-    if (coordinate().latitude() == 0 && coordinate().longitude() == 0) {
```

Comment:
This is incorrect. lat=0, lon=0 is a valid coordinate.

Reply:
The other thing to know is that at times for various commands ArduPilot (but not PX4) can use lat=0, lon=0 to indicate a lack of position in a command. So you may see other cases of this.

Reply:
True, but do we really want to check whether long and lat is zero (we want to skip processing "null island"?) Or it's just a wrong way to catch invalid coordinate.

Reply:
In this specific case there was some transient state which was causing it. I don't quite remember the details. Without going back through commit history to try to find those details it's best to leave it alone and assume the semantics are correct. As opposed to changing it and possibly breaking something in a subtle way.

Reply:
I see, is it worth to add a comment about that to avoid confusion?

Reply:
Other than what is already there not sure what else is known to be added